### PR TITLE
Simplify scanning implementation

### PR DESCRIPTION
**Pull request by Cierpliwy**
Scanning observable is currently able to delay scan subscription when other one is in progress and due to different parameters cannot be shared. Previously subscription queue was subscribing and disposing on specified thread. This caused an issue where next scan subscription could happen before disposal of previous one even when from user perspective order was different.

I decided to remove subscribing on different thread as library is not guaranteed to be thread safe at all. This commit fixes case when start -> stop scan cycle performed quickly causes observable to not emit scan results.

### CODE REVIEW

File: Source/BluetoothManager.swift [hunk 1]
```diff
@@ -70,11 +70,8 @@ public class BluetoothManager {
     /// Creates new `BluetoothManager` instance with specified implementation of `RxCentralManagerType` protocol which will be
     /// used by this class. Most of a time `RxCBCentralManager` should be chosen by the user.
     /// - parameter centralManager: Implementation of `RxCentralManagerType` protocol used by this class.
-    /// - parameter queueScheduler: Scheduler on which all serialised operations are executed (such as scans). By default main thread is used.
-    init(centralManager: RxCentralManagerType,
-         queueScheduler: SchedulerType = ConcurrentMainScheduler.instance) {
+    init(centralManager: RxCentralManagerType) {
```
Comment (dariuszseweryn) [inline Source/BluetoothManager.swift]: Update comment in line 103 of the current file regarding the init method.

File: Source/Observable+QueueSubscribeOn.swift [hunk 2]
```diff
@@ -25,51 +25,39 @@ import RxSwift
...
         queue.append(observable)
         if execute {
             // Observable is scheduled immidiately
-            queue.first?.delayedSubscribe(on: scheduler)
+            queue.first?.delayedSubscribe()
         }
     }
 
     func unsubscribe(observable: DelayedObservableType) {
-        lock.lock(); defer { lock.unlock() }
```
Comment (dariuszseweryn) [inline Source/Observable+QueueSubscribeOn.swift]: Are you sure that unsubscribing two scans from different threads will not change the `index` of the `observable`? Unlikely but seems possible

Comment (dariuszseweryn) [reply Source/Observable+QueueSubscribeOn.swift]: Queueing seems to need synchronization too

File: Source/BluetoothManager.swift (hunk 1 above)
Comment (Cierpliwy) [reply Source/BluetoothManager.swift]: Done :)

**Pull request merged**